components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/datasource/configreader/ConfigReader.java

```diff
@@ -67,6 +68,21 @@
         return new HashMap<>();
     }
 
+    private static Map<String, Object> readCarbonConfigs() {
+        try {
+            return (Map<String, Object>) DataHolder.getInstance().getConfigProvider()
+                    .getConfigurationObject(CARBON_NAMESPACE);
+        } catch (Exception e) {
+            log.error("Failed to read  deployment.yaml file . ", e);
+        }
+        return new HashMap<>();
+    }
+
+    public String getSolutionType() {
+        Object solutionType = carbonConfigs.get("type");
+        return solutionType != null ? solutionType.toString() : "sp";
```

Comment:
Use constants for "sp", "type", "admin", etc... and try to re-use them in other places as well

Reply:
Fixed by b14dd18f396f799bd32670842869feb2475e778e